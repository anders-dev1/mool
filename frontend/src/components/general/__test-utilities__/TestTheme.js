export const testTheme = {
    textColor: "black",
    primaryColor: "#007BFF",
    primaryHover: "rgba(0,123,255,0.33)",
    successColor: "#28A745",
    errorColor: "#F02849",
    backgroundColor: "#F0F2F5",
    buttonTextColor: "#FFFFFF",
    dividerColor: "#C7CBD1",
    inputColor: "#99A5B0",
    spinnerColor: "#FFFFFF transparent transparent transparent",
    font: "Acumin Pro, Arial, sans-serif !important;"
};