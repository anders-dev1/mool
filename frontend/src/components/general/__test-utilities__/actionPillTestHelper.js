import {screen} from "@testing-library/react";

export function findActionPillByIcon(icon) {
    return screen.getByText(icon).closest('div');
}

export function findActionPillNumberByIcon(icon) {
    const pill = findActionPillByIcon(icon);
    const numberP = pill.lastChild.lastChild;
    return numberP.textContent;
}