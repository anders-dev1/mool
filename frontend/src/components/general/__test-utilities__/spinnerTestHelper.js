export function findSpinner(container){
    return container.getElementsByClassName('lds-ring')[0];
}