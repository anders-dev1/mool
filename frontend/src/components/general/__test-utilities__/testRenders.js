import React from 'react'
import {render} from '@testing-library/react'
import {Provider} from 'react-redux'
import {setupStore, store} from "../../../redux/store";
import {testTheme} from "./TestTheme";
import {ThemeProvider} from "styled-components";

// This is used to set up the store for tests.
export function renderWithProviders(
    ui,
    {
        preloadedState = {},
        // Automatically create a store instance if no store was passed in
        testStore = setupStore(preloadedState),
        ...renderOptions
    } = {}
) {
    function Wrapper({children}) {
        return(
            <Provider store={testStore}>
                <ThemeProvider theme={testTheme}>
                {children}
                </ThemeProvider>
            </Provider>
        );
    }

    return {testStore, ...render(ui, {wrapper: Wrapper, ...renderOptions})}
}

// This is used with the react-test-renderer to render dom in tests.
export function internalsForRenderer(
    ui,
    {
        preloadedState = {},
        testStore = setupStore(preloadedState)
    } = {}
){
    function Wrapper(children) {
        return(
            <Provider store={testStore}>
                <ThemeProvider theme={testTheme}>
                    {children}
                </ThemeProvider>
            </Provider>
        );
    }

    return Wrapper(ui);
}