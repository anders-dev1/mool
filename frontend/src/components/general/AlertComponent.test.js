import renderer from "react-test-renderer";
import {ThemeProvider} from "styled-components";
import {testTheme} from "./__test-utilities__/TestTheme";
import Alert from "./AlertComponent";
import {fireEvent, render} from "@testing-library/react";

test('renders correctly', () => {
    const tree = renderer.create(
        <ThemeProvider theme={testTheme}>
            <Alert
                className={"class"}
                header={"this is the header text"}
                text={"this is the alert text"}/>
        </ThemeProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});

test("click on close button, when hideCallBack is supplied, should call hideCallBack", () =>{
    // Arrange
    const mockCallBack = jest.fn();
    const {container} = render(
        <ThemeProvider theme={testTheme}>
            <Alert hideCallback={mockCallBack}/>
        </ThemeProvider>);
    const closeButton = container.getElementsByClassName("ml-auto");

    // Act
    fireEvent.click(closeButton[0]);

    // Assert
    expect(mockCallBack).toHaveBeenCalled();
});

describe("when given color", () => {
    function renderWithColor(color) {
        const {container} = render(
            <ThemeProvider theme={testTheme}>
                <Alert color={color}/>
            </ThemeProvider>);

        return container.querySelector('div');
    }

    test("undefined, should display the primary theme color", () => {
        // Arrange
        const alert = renderWithColor();

        // Assert
        expect(alert).toHaveStyle(`background-color: ${testTheme.primaryColor};`);
    });

    test("primary, should display the primary theme color", () => {
        // Arrange
        const alert = renderWithColor("primary");

        // Assert
        expect(alert).toHaveStyle(`background-color: ${testTheme.primaryColor};`);
    });

    test("success, should display the primary theme color", () => {
        // Arrange
        const alert = renderWithColor("success");

        // Assert
        expect(alert).toHaveStyle(`background-color: ${testTheme.successColor};`);
    });

    test("error, should display the primary theme color", () => {
        // Arrange
        const alert = renderWithColor("error");

        // Assert
        expect(alert).toHaveStyle(`background-color: ${testTheme.errorColor};`);
    });
});