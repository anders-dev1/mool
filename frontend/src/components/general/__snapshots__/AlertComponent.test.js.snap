// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bczRLJ hZqhXJ class"
>
  <div
    className="flex fill"
  >
    <p
      className="sc-gsnTZi iuwaYe"
    >
      this is the header text
    </p>
    <svg
      className="sc-hKMtZM rlYum ml-auto"
      height="24"
      onClick={[Function]}
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2c5.514 0 10 4.486 10 10s-4.486 10-10 10-10-4.486-10-10 4.486-10 10-10zm0-2c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm6 16.094l-4.157-4.104 4.1-4.141-1.849-1.849-4.105 4.159-4.156-4.102-1.833 1.834 4.161 4.12-4.104 4.157 1.834 1.832 4.118-4.159 4.143 4.102 1.848-1.849z"
      />
    </svg>
  </div>
  <p
    className="sc-dkzDqf ixrmBA"
  >
    this is the alert text
  </p>
</div>
`;
